src/main/scala/edu/berkeley/ce/rockslicing/RockSlicer.scala

```diff
@@ -47,14 +50,65 @@ object RockSlicer {
       Block(center, block.nonRedundantFaces)
     }
 
-    // Calculate centroid of each block
-    val centroidBlocks = nonRedundantBlocks.map { block =>
+    // Find all blocks that contain processor joints
+    val processorBlocks = nonRedundantBlocks.filter { block => 
+      block.faces.exists { face => face.processorJoint}
+    }
+
+    // Find blocks that do not contain processor joints
+    val realBlocks = nonRedundantBlocks.filter { block =>
+      val faceFlags = block.faces map { face =>
+        face.processorJoint
+      }
+      !faceFlags.contains(true)
+    }
```

Comment:
Actually, Scala should have a `partition` method for `Seq` that can split it into these two collections for you, and it will be clearer and more concise that way.